lib/api/repository/oauth_repository.dart

```diff
@@ -18,17 +19,27 @@ class OauthRepositoryImpl extends OauthRepository {
   OauthRepositoryImpl(this._oauthService);
 
   @override
-  Future<OauthTokenResponse> login({
+  Future<OAuthToken> login({
     required String email,
     required String password,
-  }) {
-    return _oauthService
-        .login(OauthTokenRequest(
-          email: email,
-          password: password,
-          clientId: Env.basicAuthClientId,
-          clientSecret: Env.basicAuthClientSecret,
-        ))
-        .then((response) => response.data.attributes);
+  }) async {
+    try {
+      return await _oauthService
+          .login(OauthTokenRequest(
+            email: email,
+            password: password,
+            clientId: Env.basicAuthClientId,
+            clientSecret: Env.basicAuthClientSecret,
+          ))
+          .then((baseResponse) => baseResponse.data.attributes)
+          .then((response) => OAuthToken(
+                accessToken: response.accessToken,
+                refreshToken: response.refreshToken,
+                tokenType: response.tokenType,
+                expiresIn: response.expiresIn,
+              ));
+    } catch (exception) {
```

Comment:
I would reccommend not to mix chaining way and the async await way. How about:
try{ 
final response = await oauthService.login(...);
final data= response.data.attr;
return OAuthToken();
} catch (ex) {
  throw Exception
}

Reply:
or chaining use onError

Reply:
why shouldn't we keep them in a chain?

Reply:
alright, got the point here 👌

Reply:
Fixed